pkg/helm/chart_info.go

```diff
@@ -146,6 +146,25 @@ func parseChartRequirementsFile(chartDirectory string, apiVersion string) (Chart
 	return chartRequirements, nil
 }
 
+func removeIgnored(rootNode *yaml.Node, rootKind yaml.Kind) {
+	var toDelete []int
+	for i, node := range rootNode.Content {
+		if strings.Contains(node.HeadComment, "# -- @ignore") {
+			toDelete = append(toDelete, i)
+			if rootKind == yaml.MappingNode {
+				toDelete = append(toDelete, i+1)
+			}
+		}
+	}
+	for i := len(toDelete) - 1; i >= 0; i-- {
+		var d int = toDelete[i]
+		rootNode.Content = append(rootNode.Content[:d], rootNode.Content[d+1:]...)
+	}
+	for _, node := range rootNode.Content {
+		removeIgnored(node, node.Kind)
+	}
```

Comment:
So I think this can be simplified.

```go
newContent := make([]*yaml.Node , 0, len(rootNode.Content))

for _, node := range rootNode.Content {
	if !strings.Contain(node.HeadComment, "@ignore") {
		removeIgnored(node, node.Kind)
		newContent = append(newContent, node)
	}
}

rootNode.Content = newContent
```

Reply:
Did you test it? I thought that deleting an item during iteration would break iteration order.

Reply:
Also - when `Kind` of nodes root is `yaml.MappingNode` you have to delete two consecutive items